src/ebmc/ranking_function.cpp

```diff
@@ -114,51 +85,79 @@ ebmc_baset::propertyt &ranking_function_checkt::find_property()
   UNREACHABLE;
 }
 
-int ranking_function_checkt::operator()()
+int do_ranking_function(
+  const cmdlinet &cmdline,
+  message_handlert &message_handler)
 {
   // get the transition system
-  int exit_code = get_transition_system(
-    cmdline, message.get_message_handler(), transition_system);
+  transition_systemt transition_system;
+
+  int exit_code =
+    get_transition_system(cmdline, message_handler, transition_system);
 
   if(exit_code != -1)
     throw ebmc_errort();
 
   CHECK_RETURN(transition_system.trans_expr.has_value());
 
   // parse the ranking function
-  const auto ranking_function = parse_ranking_function();
+  if(!cmdline.isset("ranking-function"))
+    throw ebmc_errort() << "no candidate ranking function given";
```

Comment:
Is this really the right place for performing this check? I'd expect an invariant here for whatever code is to invoke `do_ranking_function` must only do so when this argument is present.

Reply:
That is `is_ranking_function` -- the code in `do_ranking_function` could move to ebmc_parse_options.cpp.